File: backend/Models/DTO/Responses/DateValidationResponse.cs
namespace MedicalDemo.Models.DTO.Responses;

public class DateValidationResponse : GenericResponse
{
    private DateValidationResponse() { }

    public ViolationResultResponse? ViolationResultResponse { get; set; }

    public static DateValidationResponse NonViolationFailure(string message)
    {
        return new DateValidationResponse
        {
            Success = false,
            Message = message,
            ViolationResultResponse = null
        };
    }

    public static DateValidationResponse ViolationFailure(ViolationResultResponse result, string message)
    {
        return new DateValidationResponse
        {
            Success = false,
            Message = message,
            ViolationResultResponse = result
        };
    }

    public static DateValidationResponse NoViolations()
    {
        return new DateValidationResponse
        {
            Success = true,
            Message = "",
            ViolationResultResponse = null
        };
    }
}

File: frontend/src/lib/models/DateValidationResponse.ts
export interface ConstraintResultResponse {
  isViolated: boolean;
  message: string | null;
  isOverridable: boolean | null;
}

export interface ViolationResultResponse {
  isViolation: boolean;
  isOverridable: boolean | null;
  violations: ConstraintResultResponse[];
}

export interface DateValidationResponse {
  success: boolean;
  message: string;
  violationResultResponse: ViolationResultResponse | null;
}